### ISSUE TITLE
should anything() matcher swallow subsequent arity?

### ISSUE THREAD

**Issue by searls** [testdouble/testdouble.js#32]
Try this:

```
drive = td.create()
td.when(drive(5, anything())).thenReturn('A')

drive()  //undefined, expected
drive(5)  //'A'
drive(5, 'hi') //'A'
drive(5, 'hi', 'bye')  //'A'
drive(6) // undefined, expected
drive(6, null) //undefined, expected
```

I'm torn on what `drive(5)` and `drive(5, 'hi', 'bye')` should do in this case. Right now `anything()` will slurp up all remaining args to just match any number, which isn't as strict as I'd personally like but would make it a bridge less far from sinon/jasmine's default-to-unconditional-stubbing.

Thoughts?  @jasonkarns

**Comment by searls** [comment]
I bet however we fix #33 will incidentally change the above behavior.

Wish I had a stronger opinion for how to deal with cases when people don't care about extra arity, but I'm fine defaulting to a strict one since this is for isolated unit testing

**Comment by searls** [comment]
As expected, 0.3.1 / #33 drastically changed the behavior of stubbing. I probably should have minor-bumped, but on further reflection it was a bug. Yay semver for encoding my opinions on what counts as a bug!

**Comment by searls** [comment]
New behavior, which I think is "correct":

```> drive = td.create()
{ [Function] toString: [Function] }
> td.when(drive(5, anything())).thenReturn('A')
{ [Function] toString: [Function] }
> 
> drive()  //undefined, expected
undefined
> drive(5)  //'A'
undefined
> drive(5, 'hi') //'A'
'A'
> drive(5, 'hi', 'bye')  //'A'
undefined
> drive(6) // undefined, expected
undefined
> drive(6, null) //undefined, expected
undefined
```

**Comment by searls** [comment]
Closed in favor of implementing #36
